src/render/painter.js

```diff
@@ -293,19 +293,27 @@ class Painter {
         this.gl.depthRange(nearDepth, farDepth);
     }
 
-    translatePosMatrix(matrix, tile, translate, anchor) {
+    translatePosMatrix(matrix, tile, translate, anchor, inPixelUnits) {
         if (!translate[0] && !translate[1]) return matrix;
 
-        if (anchor === 'viewport') {
-            const sinA = Math.sin(-this.transform.angle);
-            const cosA = Math.cos(-this.transform.angle);
+        const angle = inPixelUnits ?
```

Comment:
It's not immediately obvious here what `inPixelUnits` has to do with the anchor angle -- I had to go back and look at the use of `getGlCoordMatrix` in `draw_symbol` to puzzle it out. Maybe just an explanatory comment would help?